use_cases/eluc/app/app.py

```diff
@@ -1,118 +1,49 @@
-
-app.layout = html.Div([
-    dcc.Markdown('''
-# Land Use Optimization
-This site is for demonstration purposes only.
-
-For a given context cell representing a portion of the earth,
-identified by its latitude and longitude coordinates, and a given year:
-* What changes can we make to the land usage
-* In order to minimize the resulting estimated CO2 emissions? (Emissions from Land Use Change, ELUC,
- in tons of carbon per hectare)
+context_component.register_callbacks(app)
+filter_component.register_callbacks(app)
+dms_component.register_callbacks(app)
 
-'''),
-    dcc.Markdown('''## Context'''),
-    html.Div([
-        dcc.Graph(id="map", figure=map_component.get_map_fig(), style={"grid-column": "1"}),
-        html.Div([map_component.get_context_div()], style={"grid-column": "2"}),
-        html.Div([legend_component.get_legend_div()], style={"grid-column": "3"})
-    ], style={"display": "grid", "grid-template-columns": "auto 1fr auto", 'position': 'relative'}),
-    dcc.Markdown('''## Actions'''),
-    html.Div([
-        html.Div([prescription_component.get_presc_select_div()], style={"grid-column": "1"}),
-        html.Div([chart_component.get_chart_select_div()],
-                 style={"grid-column": "2", "margin-top": "-10px", "margin-left": "10px"}),
-    ], style={"display": "grid", "grid-template-columns": "45% 15%"}),
-    html.Div([
-        html.Div(lock_component.get_checklist_div(), style={"grid-column": "1", "height": "100%"}),
-        html.Div(sliders_component.get_sliders_div(), style={'grid-column': '2'}),
-        dcc.Graph(id='context-fig',
-                  figure=chart_component.create_treemap(type_context=True),
-                  style={'grid-column': '3'}),
-        dcc.Graph(id='presc-fig',
-                  figure=chart_component.create_treemap(type_context=False),
-                  style={'grid-clumn': '4'})
-    ], style={'display': 'grid', 'grid-template-columns': '4.5% 40% 1fr 1fr', "width": "100%"}),
-    # The above line can't be set to auto because the lines will overflow!
-    html.Div([
-        sliders_component.get_frozen_div(),
-        html.Button("Sum to 100%", id='sum-button', n_clicks=0),
-        html.Div(id='sum-warning')
-    ]),
-    dcc.Markdown('''## Outcomes'''),
-    prediction_component.get_predict_div(),
-    dcc.Markdown('''## Trivia'''),
-    trivia_component.get_trivia_div(),
-    dcc.Markdown('''## References'''),
-    references_component.get_references_div()
-], style={'padding-left': '10px'},)
+app.layout = html.Div(
+    children=[
+        intro_component.get_div(),
+        context_component.get_div(),
+        filter_component.get_div(),
+        dms_component.get_div(),
+        references_component.get_references_div()
+    ]
+)
 
 if __name__ == '__main__':
-    app.run_server(host='0.0.0.0', debug=False, port=4057, use_reloader=False, threaded=False)
+    app.run_server(host='0.0.0.0', debug=False, port=4057, use_reloader=True, threaded=True)
```

Comment:
Did you mean to leave user_reloader to True?

Reply:
According to https://dash.plotly.com/devtools you're not changing the code in the deployed app, and it makes it slower to start, so I'll leave it to False